### ISSUE TITLE
[BUG] Plugin not picking up certain flashcards

### ISSUE THREAD

**Issue by bgreenawald** [st3v3nmw/obsidian-spaced-repetition#915]
**Describe the bug**
The plugin suddenly only picks up ~half of the flashcards in my Vault. As of a few days ago, I had nearly 500 flashcards, but when I loaded the flashcard review today, it only showed that I had around 250 flashcards. I can confirm that all the other flashcards are still in the Vault and found an example of a file with flashcards past their review date that weren't being picked up at all. I currently don't see any pattern between what it picks up and what it doesn't.

**Versions (please complete the following information):**

-   OS: Windows 11, Android
-   Obsidian version: 1.5.11
-   Plugin version: 1.12.1
-   If on desktop, Installer version: 1.4.13

**Comment by ronzulu** [comment]
Hi @bgreenawald 

Thanks for letting us know.

Could you please post (as attachment, not copy/paste) a markdown file that contains flashcard(s) that don't appear in the review modal.

Also, please attach the plugin's settings file, located e.g.
`D:\Obsidian\.obsidian\plugins\obsidian-spaced-repetition\data.json`

Cheers
Ronny

**Comment by bgreenawald** [comment]
Hi Ronny,

Thanks for the quick response. Attached is an example file that isn't showing up. Here is a picture of the current review screen:
![image](https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/8428972/7eb5a242-e9ae-4034-b206-994d0237e29f)

Best,
Ben

[Avicenna.md](https://github.com/st3v3nmw/obsidian-spaced-repetition/files/14729355/Avicenna.md)
[data.json](https://github.com/st3v3nmw/obsidian-spaced-repetition/files/14729370/data.json)

**Comment by Garg0yle** [comment]
I'm having the same issue all of a sudden. From about 50-60 flashcards in my vault to 2. Here are attachments. Thanks for your work on this plugin btw

[The Taliban at war_ 2001-2018.md](https://github.com/st3v3nmw/obsidian-spaced-repetition/files/14733369/The.Taliban.at.war_.2001-2018.md) 

[data.json](https://github.com/st3v3nmw/obsidian-spaced-repetition/files/14733377/data.json)

**Comment by Yury-Dictum** [comment]
Hi! Same problem here. Two days ago there used to be around 130 cards in the queue, now there are none.

[3. 🚮 giving smb a hard time.md](https://github.com/st3v3nmw/obsidian-spaced-repetition/files/14741164/3.giving.smb.a.hard.time.md)

[data.json](https://github.com/st3v3nmw/obsidian-spaced-repetition/files/14741178/data.json)

**Comment by ronzulu** [comment]
Thanks, I can reproduce the problem now, hopefully fixed tomorrow

**Comment by Scicily** [comment]
I'm having the same issue all of a sudden.

**Comment by ronzulu** [comment]
Hi all

Beta release is now available.

There is a file called `main.js` in this zip file
[main_915A.zip](https://github.com/st3v3nmw/obsidian-spaced-repetition/files/14751047/main_915A.zip)


You will need to copy that to the obsidian plug-in folder, e.g.
`D:\Obsidian\Obsidian\.obsidian\plugins\obsidian-spaced-repetition`

Then reload the plug-in, or restart obsidian. 

Any feedback appreciated.

Ronny

**Comment by Effeh** [comment]
No change for me with the beta version. I use the "convert ==highlights== to clozes" function, and created 20 flashcards earlier today, but none of them are getting picked up even with the new version

**Comment by ronzulu** [comment]
@Effeh 

Could you please post (as attachment, not copy/paste) a markdown file that contains flashcard(s) that don't appear in the review modal.

Also, please attach the plugin's settings file, located e.g.
`D:\Obsidian\.obsidian\plugins\obsidian-spaced-repetition\data.json`

**Comment by lukeh2313** [comment]
I am also having the same issue, only one deck of my 7-8 decks are showing up with no change after trying your beta version. 

Edition	Windows 10 Home
OS build	19045.4170

Obsidian version: 1.5.11

Plugin version: 1.12.1

Example note that doesn't show up: [dividends vs share repurchases.md](https://github.com/st3v3nmw/obsidian-spaced-repetition/files/14756993/dividends.vs.share.repurchases.md)

Data: [data.json](https://github.com/st3v3nmw/obsidian-spaced-repetition/files/14757000/data.json)

Thanks for the support!

**Comment by Effeh** [comment]
Requested files for @ronzulu

[This Is My Body.md](https://github.com/st3v3nmw/obsidian-spaced-repetition/files/14757039/This.Is.My.Body.md)
[data.json](https://github.com/st3v3nmw/obsidian-spaced-repetition/files/14757046/data.json)

**Comment by ronzulu** [comment]
Hi @Effeh 

Your issue may appear the same as others on this topic, but actually is different.

There is a blank line at the very start of your file, i.e. line (1) in the following:

![image](https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/75528127/19fec756-293b-4661-8618-ee0140c3b628)

This means that the `tags: Chinese` is not recognized by Obsidian as frontmatter.

Simply remove that blank line, and it should work.

After deleting the blank line, Obsidian recognizes this as frontmatter and renders that section in a special manner

![image](https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/75528127/4ca37050-0a99-4a60-801f-b0cde1457dec)

**Comment by ronzulu** [comment]
Hi @lukeh2313 

There were significant code changes in plugin version 1.12 to support:
- [Multiple Tags Within a Single File](https://www.stephenmwangi.com/obsidian-spaced-repetition/flashcards/#multiple-tags-within-a-single-file)
- [A Single Card Within Multiple Decks](https://www.stephenmwangi.com/obsidian-spaced-repetition/flashcards/#a-single-card-within-multiple-decks)

It didn't occur to me that someone might place the topic tag on the line immediately following the last line of a question, and I didn't code for it.

For example, you have:
```
Why might a company prefer to repurchase shares over the issuance of a dividend?
?
1. Avoids double taxation, first at a corporate level (dividends are not tax deductible) then at individual income tax level when dividends are distributed.
2. Share repurchases inflate EPS by reducing number of shares outstanding which may increase company share price.
3. Share repurchases signal to the market that management believes its shares are undervalued, which might be seen as a good signal.
4. Once dividends are started, they are typically expected to continue. Stopping dividend payments is seen very poorly by the markets and are interpreted as a sign of lower future profits.
#wsp-red <!--SR:!2024-03-26,1,210-->
```

It should work if you move the topic tag `#wsp-red` to before the question, e.g.:

```
#wsp-red
Why might a company prefer to repurchase shares over the issuance of a dividend?
?
1. Avoids double taxation, first at a corporate level (dividends are not tax deductible) then at individual income tax level when dividends are distributed.
2. Share repurchases inflate EPS by reducing number of shares outstanding which may increase company share price.
3. Share repurchases signal to the market that management believes its shares are undervalued, which might be seen as a good signal.
4. Once dividends are started, they are typically expected to continue. Stopping dividend payments is seen very poorly by the markets and are interpreted as a sign of lower future profits.
<!--SR:!2024-03-26,1,210-->
```

Cheers
Ronny

**Comment by bgreenawald** [comment]
Beta version fixed the issue for me.

Best,
Ben

**Comment by Scicily** [comment]
Thx for your help.
I'll try it latter. (mac OS, it's hard to find the file to me...)

Scicily

Effeh ***@***.***> 於 2024年3月26日 週二 下午6:43寫道：

**Comment by lukeh2313** [comment]
Thanks Ronny, that is working now!

**Comment by Effeh** [comment]
Thanks Ron. Sorry about the confusion. I'm not sure how that happened---I've been using the same template file for quite a while, and the problem only cropped up very recently. All fixed now regardless

**Comment by huetruong** [comment]
I'm just coming here because all my cards doesn't show up even in 1.12.2.

Here's an example

[AZ305ST1-Q60.md](https://github.com/st3v3nmw/obsidian-spaced-repetition/files/14812721/AZ305ST1-Q60.md)

**Comment by Scicily** [comment]
same me! QQ

Hue Truong ***@***.***> 於 2024年3月31日 週日 上午1:03寫道：

**Comment by ronzulu** [comment]
Hi @huetruong thanks for posting the markdown file. 

Your specific issue is being addressed under [[BUG] Unable to recognize multi-line card that begins immediately after the frontmatter's closing line](https://github.com/st3v3nmw/obsidian-spaced-repetition/issues/922)

@Scicily There might be multiple reasons why you are experiencing this. Please post a markdown file that exhibits the problem (attach, don't copy/paste)

Cheers
Ronny

**Comment by huetruong** [comment]
@ronzulu Please don't strip away my special character I used to separate the paragraphs. The last time I used a space, " ", in a blank line, it was strip away with strip() to remove whitespace.

**Comment by ronzulu** [comment]
> Please don't strip away my special character I used to separate the paragraphs. The last time I used a space, " ", in a blank line, it was strip away with strip() to remove whitespace.

Hi @huetruong, is your comment related to this issue? If so, please give further explanation.

**Comment by Scicily** [comment]
Hello,

I am currently facing an issue where I cannot display the review cards
correctly. Since I am using a Mac Sonoma version 14.4.1, I am not quite
sure how to locate the MD file you mentioned. If possible, could you please
instruct me on how to proceed? Thank you very much.

Sincerely Scicily

Ronny Zulaikha ***@***.***> 於 2024年3月31日 週日 下午6:54寫道：

**Comment by huetruong** [comment]
> @ronzulu Please don't strip away my special character I used to separate the paragraphs. The last time I used a space, " ", in a blank line, it was strip away with strip() to remove whitespace.

A few months ago, a trim function was introduced which broke my flashcards. Currently, I am using a zero-width character to prevent the trim function from deleting my intentionally added blank lines, which make the flashcards easier to read. I am requesting that you take note of this issue to ensure that it does not occur again in the future. I have attached an example that illustrates the way I use the zero width character.

![az305st1-q60](https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/68408963/4ae5238c-8f61-4a41-8a52-853c9d3c8e6b)

**Comment by ronzulu** [comment]
Hi @huetruong I would suggest you create a new discussion item, as this is not quite on topic.

I can put a comment in the code about this, please let us know what is the zero width character you are using.

longer term I think the solution will be something like:
[[FEAT] Support multiline questions with blank lines in both the card front and back](https://github.com/st3v3nmw/obsidian-spaced-repetition/issues/629)

Cheers
Ronny

**Comment by ronzulu** [comment]
> Hello, I am currently facing an issue where I cannot display the review cards correctly. Since I am using a Mac Sonoma version 14.4.1, I am not quite sure how to locate the MD file you mentioned. If possible, could you please instruct me on how to proceed? Thank you very much. Sincerely Scicily

Hi @Scicily 

Sorry, I'm not a Mac person - don't know where Obsidian. As an idea, click on the "vault" icon. On a PC it looks like:
![image](https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/75528127/b72efab7-9e24-4de5-9b08-f4af187a6ae6)

On the left hand side it shows the location on the vault in the file system, perhaps on your Mac it would give you a clue:
![image](https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/75528127/a066caef-06b3-4678-bd0f-f4dfa299b2be)

Also, when you say "cannot display the review cards correctly" is the card:
1. Missing, or
2. Present and does display, but displays incorrectly

If the card does display, but incorrectly, then please open a new issue, as this issue is for cards that have gone missing after the recent release.

Cheers
Ronny

**Comment by huetruong** [comment]
> Hi @huetruong I would suggest you create a new discussion item, as this is not quite on topic.
> 
> I can put a comment in the code about this, please let us know what is the zero width character you are using.
> 
> longer term I think the solution will be something like: [[FEAT] Support multiline questions with blank lines in both the card front and back](https://github.com/st3v3nmw/obsidian-spaced-repetition/issues/629)
> 
> Cheers Ronny

I would definitely love to have that in the long term. Here is what I am using below as a work around.

https://www.groovypost.com/howto/type-zero-width-space/

**Comment by ronzulu** [comment]
Hi all

Beta release is now available. I've tested with most (if not all) of the files posted above and seems ok.

There is a file called `main.js` in this zip file
[main_915B.zip](https://github.com/st3v3nmw/obsidian-spaced-repetition/files/14844589/main_915B.zip)

You will need to copy that to the obsidian plug-in folder, e.g.
`D:\Obsidian\.obsidian\plugins\obsidian-spaced-repetition`

Then reload the plug-in, or restart obsidian. 

Any feedback appreciated.

Ronny

**Comment by Scicily** [comment]
Thank you for your assistance. In the meantime, I will research the data
storage locations in MacOS and then apply the information you provided. (I
am a new user of MacOS, still figuring out the logic of using a Mac.) Thank
you. Sincerely, Scicily


Ronny Zulaikha ***@***.***> 於 2024年4月3日 週三 上午7:29寫道：

**Comment by Garg0yle** [comment]
> Hi all
> 
> Beta release is now available. I've tested with most (if not all) of the files posted above and seems ok.
> 
> There is a file called `main.js` and `style.css` in this zip file [main_915C.zip](https://github.com/st3v3nmw/obsidian-spaced-repetition/files/14878031/main_915C.zip)
> 
> You will need to copy them to the obsidian plug-in folder, e.g. `D:\Obsidian\.obsidian\plugins\obsidian-spaced-repetition`
> 
> Then reload the plug-in, or restart obsidian.
> 
> Any feedback appreciated.
> 
> Ronny

Hi Ronny, this fixed the issue for me! Thanks so much for the work here.

**Comment by huetruong** [comment]
> Hi all
> 
> Beta release is now available. I've tested with most (if not all) of the files posted above and seems ok.
> 
> There is a file called `main.js` and `style.css` in this zip file [main_915C.zip](https://github.com/st3v3nmw/obsidian-spaced-repetition/files/14878031/main_915C.zip)
> 
> You will need to copy them to the obsidian plug-in folder, e.g. `D:\Obsidian\.obsidian\plugins\obsidian-spaced-repetition`
> 
> Then reload the plug-in, or restart obsidian.
> 
> Any feedback appreciated.
> 
> Ronny

I can confirm my issue is fixed.

**Comment by Scicily** [comment]
Thanks, I'll check it out!

Hue Truong ***@***.***> 於 2024年4月6日 週六 上午8:12寫道：

**Comment by ronzulu** [comment]
Fixed in release 1.12.4, closing this now.

**Comment by Scicily** [comment]
Repaired and fully functional. Thank you very much.

Ronny Zulaikha ***@***.***> 於 2024年4月23日 週二 下午8:14寫道：
